src/dotnetCampus.FileDownloader/SegmentFileDownloader.cs

```diff
@@ -88,23 +92,36 @@ private async Task<(WebResponse response, long contentLength)> GetContentLength(
         {
             _logger.LogInformation("开始获取整个下载长度");
 
-            // 如果用户没有说停下，那么不断下载
+            var response = await GetWebResponseAsync();
+
+            if (response == null)
+            {
+                return default;
+            }
+
+            var contentLength = response.ContentLength;
+
+            _logger.LogInformation(
+                $"完成获取文件长度，文件长度 {contentLength} {contentLength / 1024}KB {contentLength / 1024.0 / 1024.0:0.00}MB");
+
+            return (response, contentLength);
+        }
 
+        private async Task<WebResponse> GetWebResponseAsync(Action<HttpWebRequest> action = null)
```

Comment:
```suggestion
        private async Task<WebResponse?> GetWebResponseAsync(Action<HttpWebRequest> action = null)
```

不建议随便返回 null。所以还是要充分利用好可空类型。